### PR TITLE
[Enhancement] (nereids)implement RecoverDatabaseCommand in nereids

### LINKED ISSUE #42544: [Enhancement] (nereids)implement RecoverDatabaseCommand in nereids
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

RecoverDatabaseCommand is supported in legacy planner but not in nereids(new doris fe planner). you need implement it in nereids.The DorisParser.g4 has the syntax definition already(see bellow). And you can check RecoverDbStmt.java in legacy planner to see what the command does and migrate it to nereids planner
```unsupportedRecoverStatement : RECOVER DATABASE name=identifier id=INTEGER_VALUE? (AS alias=identifier)?```

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Vallishp**
## Proposed changes

Issue Number: close #42544 

support recover database command in nereids

### CODE REVIEW

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName,
+                        PrivPredicate.ALTER_CREATE)) {
+            ErrorReport.reportAnalysisException(
+                    ErrorCode.ERR_DBACCESS_DENIED_ERROR, ctx.getQualifiedUser(), dbName);
+        }
+
+        RecoverDbStmt recoverStmt = new RecoverDbStmt(dbName, dbId, newDbName);
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:85]: pls call the new recoverDatabase(String dbName, long dbId, String newDbName) function, and remove dependence of  RecoverDbStmt

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 2]
```diff
@@ -601,6 +601,63 @@ public void replayDropDb(String dbName, boolean isForceDrop, Long recycleTime) t
...
+            MetaLockUtils.writeUnlockTables(tableList);
+            db.writeUnlock();
+            unlock();
+        }
+        LOG.info("recover database[{}]", db.getId());
+    }
+
     public void recoverDatabase(RecoverDbStmt recoverStmt) throws DdlException {
         // check is new db with same name already exist
         String newDbName = recoverStmt.getNewDbName();
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:663]: you can call the new recoverDatabase(String dbName, long dbId, String newDbName) function to remove duplicate code

Comment (Vallishp) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java]: thanks have handled it

**Comment by Vallishp** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

File: regression-test/suites/ddl_p0/test_recover_db.groovy [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        properties(
+            "replication_num"="1",
+            "light_schema_change"="true",
+            "compression"="zstd"
+        );
+    """
+    qt_select "show tables";
+    
+    sql " drop database $db"
+    sql " recover database $db"
```
Comment (starocean999) [inline regression-test/suites/ddl_p0/test_recover_db.groovy:71]: use checkNereidsExecute(" recover database $db") to check if the command run by nereids

File: regression-test/suites/ddl_p0/test_recover_db.groovy [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    sql " use $db"
+    qt_select1 "show tables";
+    def newdb="new_recover_db"
+    sql " drop database IF EXISTS $newdb FORCE"
+    sql " drop database $db"
+    sql " recover database $db as $newdb"
```
Comment (starocean999) [inline regression-test/suites/ddl_p0/test_recover_db.groovy:77]: use checkNereidsExecute

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * recover database command
+ */
+public class RecoverDatabaseCommand extends Command implements ForwardWithSync {
+
+    public static final Logger LOG = LogManager.getLogger(RecoverDatabaseCommand.class);
+    private String dbName;
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:42]:
```suggestion
    private final String dbName;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    private long dbId = -1;
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:43]:
```suggestion
    private final long dbId = -1;
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    private String newDbName = "";
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:44]:
```suggestion
    private final String newDbName = "";
```

**Comment by Vallishp** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
pls resolve conflicts, which we can merge it.

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
> pls resolve conflicts, which we can merge it.

handled. pls help review again. thanks

**Review by starocean999 — DISMISSED** [review]

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+     */
+
+    public RecoverDatabaseCommand(String dbName, long dbId, String newDbName) {
+        super(PlanType.RECOVER_DATABASE_COMMAND);
+        this.dbName = dbName;
+        this.dbId = dbId;
+        this.newDbName = newDbName;
+    }
+
+    public String getDbName() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:56]:
```suggestion

```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        return dbName;
+    }
+
+    public long getDbId() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:60]:
```suggestion

```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        return dbId;
+    }
+
+    public String getNewDbName() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/RecoverDatabaseCommand.java:64]:
```suggestion

```

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run p0

**Review by starocean999 — APPROVED** [review]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**